client/client.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/eigenda"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	// NOTE: this will need to be updated as plasma's implementation changes
+	decodingOffset = 2
+)
+
+// TODO: Add support for custom http client option
+type Config struct {
+	URL string
+}
+
+// ProxyClient is an interface for communicating with the EigenDA proxy server
+type ProxyClient interface {
+	Health() error
+	GetData(ctx context.Context, cert *common.Certificate, domain common.DomainType) ([]byte, error)
+	SetData(ctx context.Context, b []byte) (*common.Certificate, error)
+}
+
+// client is the implementation of ProxyClient
+type client struct {
+	cfg        *Config
+	httpClient *http.Client
+}
+
+func New(cfg *Config) ProxyClient {
+	return &client{
+		cfg,
+		http.DefaultClient,
+	}
+}
+
+// Health indicates if server is operational; useful for event based awaits
+// when integration testing
+func (c *client) Health() error {
+	url := c.cfg.URL + "/health"
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("received bad status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+// GetData fetches blob data associated with a DA certificate
+func (c *client) GetData(ctx context.Context, cert *common.Certificate, domain common.DomainType) ([]byte, error) {
+	b, err := rlp.EncodeToBytes(cert)
+	if err != nil {
+		return nil, err
+	}
+
+	// encode prefix bytes
+	b = eigenda.GenericPrefix(eigenda.Commitment(b).Encode())
```

Comment:
nit: not terribly ergonomic to have `Commitment(b)` do nothing to `b` without a corresponding call to `.Encode()`. What if it looked more like

```
type genericPrefix String
type commitment String
b = eigenda.NewGenericPrefix(eigenda.NewCommitment(b))
```

where these are functions and not type casting operators and b

Reply:
Update:

I'm now seeing that `GenericPrefix()` is already a function, but `Commitment()` is not. So my suggestion would be to make `Commitment()` into a simple function so the `Encode()` part can be dropped.